src/server/db_slice.cc

```diff
@@ -1714,10 +1691,37 @@ void DbSlice::PerformDeletion(Iterator del_it, DbTable* table) {
   PerformDeletionAtomic(del_it, exp_it, table);
 }
 
-void DbSlice::OnCbFinish() {
-  // TBD update bumpups logic we can not clear now after cb finish as cb can preempt
-  // btw what do we do with inline?
-  fetched_items_.clear();
+void DbSlice::OnCbFinishBlocking() {
+  if (IsCacheMode()) {
+    // move fetched items to local variable
+    auto moved_fetched_items_ = std::move(fetched_items_);
+    for (const auto& [key_hash, db_index, key] : moved_fetched_items_) {
+      auto& db = *db_arr_[db_index];
+
+      auto predicate = [&key](const PrimeKey& key_) { return key_ == key; };
```

Comment:
you do not comparing to `key`, you can just return true here.
The point is that it's not critical that once in a billion chance you have a collision and you bump up something else
so you can remove `key` from `fetched_items_` tuple